gobblin-temporal/src/main/java/org/apache/gobblin/temporal/joblauncher/GobblinJobLauncher.java

```diff
@@ -206,6 +206,13 @@ public void launchJob(@Nullable JobListener jobListener) throws JobException {
         cancelJob(jobListener);
       }
     } finally {
+      // NOTE: This code only makes sense when there is 1 source / workflow being launched per application. This is a stop-gap
+      // for recreating batch job behavior in GaaS. Given the current constraints of yarn applications requiring a static proxy user
+      // during application creation, it is not possible to have multiple workflows running in the same application.
+      // and so it makes sense ti just kill the job after this is complete
+      log.info("Requesting the AM to shutdown after the job {} completed", this.jobContext.getJobId());
+      eventBus.post(new ClusterManagerShutdownRequest());
```

Comment:
any consequence to doing this before checking `isLaunched`?

Reply:
Whether or not the job is launched doesn't really matter here since the design is to have a single dummy source and when that source finishes (successfully or not), we want to kill the app